src/main/java/org/folio/inventory/dataimport/handlers/actions/AbstractInstanceEventHandler.java

```diff
@@ -49,38 +47,35 @@
   protected HttpClient client;
 
   protected Future<Void> createPrecedingSucceedingTitles(Instance instance, CollectionResourceRepository precedingSucceedingTitlesRepository) {
-    Promise<Void> promise = Promise.promise();
     List<PrecedingSucceedingTitle> precedingSucceedingTitles = new ArrayList<>();
     preparePrecedingTitles(instance, precedingSucceedingTitles);
     prepareSucceedingTitles(instance, precedingSucceedingTitles);
-    List<CompletableFuture<Response>> postFutures = new ArrayList<>();
 
-    precedingSucceedingTitles.forEach(title -> postFutures.add(precedingSucceedingTitlesRepository.post(title)));
-    CompletableFuture.allOf(postFutures.toArray(new CompletableFuture<?>[]{}))
+    precedingSucceedingTitles.forEach(title -> precedingSucceedingTitlesRepository
+      .post(title)
       .whenComplete((v, e) -> {
-        if (e != null) {
-          promise.fail(e);
-          return;
+          if (e != null) {
+            LOGGER.error("Error during creating PrecedingSucceedingTitle for instance {}", e, instance.getId());
+            LOGGER.info("Error during creating PrecedingSucceedingTitles retry creating new PrecedingSucceedingTitles");
+            precedingSucceedingTitlesRepository.post(title);
+          }
         }
-        promise.complete();
-      });
-    return promise.future();
+      ));
+    return Future.succeededFuture();
   }
 
   protected Future<Void> deletePrecedingSucceedingTitles(Set<String> ids, CollectionResourceRepository precedingSucceedingTitlesRepository) {
-    Promise<Void> promise = Promise.promise();
-    List<CompletableFuture<Response>> deleteFutures = new ArrayList<>();
-
-    ids.forEach(id -> deleteFutures.add(precedingSucceedingTitlesRepository.delete(id)));
-    CompletableFuture.allOf(deleteFutures.toArray(new CompletableFuture<?>[]{}))
+    ids.forEach(id -> precedingSucceedingTitlesRepository
+      .delete(id)
       .whenComplete((v, e) -> {
-        if (e != null) {
-          promise.fail(e);
-          return;
+          if (e != null) {
+            LOGGER.error("Error during deleting PrecedingSucceedingTitles with ids {}", e, id);
+            LOGGER.info("Error during deleting PrecedingSucceedingTitles retry delete PrecedingSucceedingTitles");
+            precedingSucceedingTitlesRepository.delete(id);
+          }
         }
-        promise.complete();
-      });
-    return promise.future();
+      ));
+    return Future.succeededFuture();
```

Comment:
This code appears to finish immediately without waiting for any / all of the deletions to happen. Have I interpreted this correctly?

Reply:
Yes. For the current pub-sub implementation it is one way to avoid stacking future chains and oom. In the next implementation with kafka it can be changed